add new title for asus router firmware

derekriemer opened the pull request:
- **add new fixes for asus router firmware.**
- **update skeleton with function to update a live region temporarily.**

Commit:
add new fixes for asus router firmware.

* fixes up nav menus.
* makes some toggles labeled.
* makes tutor help offered in settings pannels accessible.

Commit:
update skeleton with function to update a live region temporarily.

jcsteh commented on framework/axSGreaseSkeleton.js:
```diff
@@ -11,6 +11,17 @@
 /*** Functions for common tweaks. ***/
 
+/**
+ * Adds text to the given live region, and clears it a second later so its no
```
```suggestion
 * Adds text to the given live region, and clears it a second later so it's no
```

jcsteh commented on framework/axSGreaseSkeleton.js:
```diff
@@ -11,6 +11,17 @@
+ * Adds text to the given live region, and clears it a second later so its no
+ * longer in the virtual buffer.
+ */
+function announce(text, region) {
```
I wonder if it's useful to either:

1. Have this function create the region if it doesn't exist yet and delay the initial announcement slightly; or
2. Add an additional function to add an off-screen live region, based on your code in init().

derekriemer replied:
please lmk if you like this approach. The promise bit is admittedly a bit weird all be it more resiliant. if you'd rather I can consolidate into 1 function with plain old set timeout.

jcsteh commented on asusrouter.user.js:
```diff
@@ -0,0 +1,245 @@
+ * Adds text to the given live region, and clears it a second later so its no
+ * longer in the virtual buffer.
+ */
+function announce(text, region) {
```
Function defined twice? :)

derekriemer replied:
fixed

jcsteh commented on asusrouter.user.js:
```diff
@@ -0,0 +1,245 @@
+});
+
+function init() {
+	const tutorHelp = document.createElement('div');
```
I don't think you need to worry about this for this PR, but this is making me think the skeleton should call a separate init function defined where the tweaks are defined so that authors don't have to modify the skeleton itself.

derekriemer replied:
I went ahead and added this, although the live region code is now more resilient.

derekriemer replied:
done

jcsteh reviewed (changes requested):
Thanks!

jcsteh commented:
Oh, please also rename this to AsusRouterA11yFixes.user.js.

derekriemer committed:
Update framework/axSGreaseSkeleton.js

Co-authored-by: James Teh <jamie@jantrid.net>

Commit:
UAddress review comments

Commit:
update readme

jcsteh commented on framework/axSGreaseSkeleton.js:
```diff
@@ -11,6 +11,59 @@
 /*** Functions for common tweaks. ***/
 
+/**
+ * Adds text to the given live region, and clears it a second later so its no
```
```suggestion
 * Adds text to the given live region, and clears it a second later so it's no
```

jcsteh commented on framework/axSGreaseSkeleton.js:
```diff
@@ -11,6 +11,59 @@
+			region.innerText = text;
+			setTimeout(() => {
+				region.innerText = '';
+			}, 1000);
```
I really wish we could convert this into an async function with awaits, but there's still no native timeout Promise, so we'd just be trading one ugliness for another. :(

jcsteh commented on framework/axSGreaseSkeleton.js:
```diff
@@ -11,6 +11,59 @@
+ * @param {string} id the name of the new live region. This is an html id.
+ * @return {!Promise<HTMLElement>} a div that contains the live region. This can typically be ignored, this exxists to aid in chaining creation of non-existant regions.
+ */
+
```
```suggestion
```

jcsteh commented on framework/axSGreaseSkeleton.js:
```diff
@@ -11,6 +11,59 @@
+		region.style.opasity = 0;
+		document.body.appendChild(region);
+		// we need to delay a little to get the new region to actually read contents.
+		// A11y api probably don't considder the relevant changes,  additions, until
```
```suggestion
		// A11y APIs probably don't treat the relevant changes as "additions" until
```

jcsteh commented on framework/axSGreaseSkeleton.js:
```diff
@@ -11,6 +11,59 @@
+		// annimation frames, so delay 134 ms to be safe.
+		setTimeout(() => {
+			resolve(region);
+		}, 134);
```
Just out of curiosity, why 134 (as opposed to 130 or similar)? Is 134 significant somehow?

derekriemer replied:
1000/30 (frames per second) is 3.333... * 3 is 333 ...4 when rounded. Ballpark math tbh.

jcsteh commented on framework/axSGreaseSkeleton.js:
```diff
@@ -53,9 +106,9 @@ function setAriaIdIfNecessary(elem) {
 }
 
-function makeElementOwn(parentElement, listOfNodes){
+function makeElementOwn(parentElement, listOfNodes) {
```
For future reference, I'd prefer to keep completely unrelated white space fixes in separate patches, just to make "real" patches easier to follow. Just leave these ones though; it's all good.

derekriemer replied:
I use vscode, which fixes indent as part of linting (alt+shift+f or format-on-save if configured). What do you use to avoid these kind of changes from happening at work? My code, all being under format-on-save, never accumulates random errors with indent like this.

jcsteh replied:
I'm very old fashioned and still use Notepad++. The Mozilla build system (and the NVDA build system) has commands to run various linters, which I always execute before submitting things. I've tried on multiple occasions to adopt VS Code, but I just can't get to a point where it helps my efficiency/productivity overall, rather than hindering it.

jcsteh commented on framework/axSGreaseSkeleton.js:
```diff
@@ -124,10 +177,13 @@ let observer = new MutationObserver(function(mutations) {
 });
 
+/** add your specific initialization here, so that if you ever update the framework from new skeleton your inits are not overridden. */
+function userInit(){}
```
This is a great idea, but I'd move it down below the `/*** Define the actual tweaks. ***/` comment so it's easier to paste in the new skeleton without breaking your stuff.

jcsteh reviewed (changes requested):
This all looks great barring some nits... but uh, where did the actual Asus Router script go? It seems to have been removed.

Commit:
add asusRouterA11yFixes which got removed after rename

derekriemer committed:
Update framework/axSGreaseSkeleton.js

Co-authored-by: James Teh <jamie@jantrid.net>

Commit:
fix position of userinit

Commit:
mirror commit from framework

Commit:
Merge branch 'master' of https://github.com/derekriemer/axSGrease

derekriemer committed:
Apply suggestions from code review

Co-authored-by: James Teh <jamie@jantrid.net>

derekriemer commented:
I can't figure out how to reply to your comment. uh ... so yeah what happened is ... I am used to using a heavily modified murcurial at work. I renamed the file and my ide at work is smart enough to rename (git mv basically) the file, so when I renamed the file in my editor, it got untracked and the new one never got re-tracked. Silly error really. This should be better.

Commit:
update apis

derekriemer commented:
This is ready for more review. I don't understand how this review thing works on github. How do I send it back to you for another go round????

jcsteh commented on AsusRouterA11yFixes.user.js:
```diff
@@ -0,0 +1,267 @@
+
+/*** Define the actual tweaks. ***/
+
+globals = [];
```
What is this for? I don't think it's used?

jcsteh commented on framework/axSGreaseSkeleton.js:
```diff
@@ -11,6 +11,59 @@
 
+/**
+ * Adds text to the given live region, and clears it a second later so it's no
+ * longer in the virtual buffer.
```
nit: This isn't specific to virtual buffers; e.g. it impacts VoiceOver just as much.

```suggestion
 * longer perceivable.
```

jcsteh commented on framework/axSGreaseSkeleton.js:
```diff
@@ -11,6 +11,59 @@
+ * @param {string} id the name of the new live region. This is an html id.
+ * @return {!Promise<HTMLElement>} a div that contains the live region. This can typically be ignored, this exxists to aid in chaining creation of non-existant regions.
+ */
+
```
```suggestion

```

jcsteh commented on framework/axSGreaseSkeleton.js:
```diff
@@ -11,6 +11,59 @@
+	return updatePromise;
+}
+
+/**
```
Um... I think this comments out a huge block of code below?

derekriemer replied:
It's just the dangling start of a sad comment that was 99% deleted. I've gone back in and eliminated the rest of it. :-)

jcsteh reviewed (changes requested):
Nearly there.

derekriemer committed:
Update framework/axSGreaseSkeleton.js

Co-authored-by: James Teh <jamie@jantrid.net>

Commit:
change buffer to perceivable

Commit:
apply rest of review suggestions

derekriemer commented:
I'm done again.

jcsteh commented on AsusRouterA11yFixes.user.js:
```diff
@@ -0,0 +1,266 @@
+	},
+];
+
+/** add your specific initialization here, so that if you ever update the framework from new skeleton your inits are not overridden. */
```
```suggestion
/** Add your specific initialization here, so that if you ever update the framework from new skeleton your inits are not overridden. */
```

jcsteh commented on framework/axSGreaseSkeleton.js:
```diff
@@ -149,5 +201,9 @@ const DYNAMIC_TWEAK_ATTRIBS = [];
 const DYNAMIC_TWEAKS = [
 ];
 
+/** add your specific initialization here, so that if you ever update the framework from new skeleton your inits are not overridden. */
```
```suggestion
/** Add your specific initialization here, so that if you ever update the framework from new skeleton your inits are not overridden. */
```

jcsteh reviewed (commented):
Thanks for your perseverance.

jcsteh committed:
Extraneous space, comment capitlisation.

jcsteh commented on AsusRouterA11yFixes.user.js:
```diff
@@ -0,0 +1,266 @@
+/**
+ * create or fetch a live region that can be used with updateLiveRegion. Returns a promise with the region.
+ * @param {string} id the name of the new live region. This is an html id.
+ * @return {!Promise<HTMLElement>} a div that contains the live region. This can typically be ignored, this exxists to aid in chaining creation of non-existant regions.
```
```suggestion
 * @return {!Promise<HTMLElement>} a div that contains the live region. This can typically be ignored, this exists to aid in chaining creation of non-existant regions.
```

jcsteh commented on framework/axSGreaseSkeleton.js:
```diff
@@ -11,6 +11,57 @@
+/**
+ * create or fetch a live region that can be used with updateLiveRegion. Returns a promise with the region.
+ * @param {string} id the name of the new live region. This is an html id.
+ * @return {!Promise<HTMLElement>} a div that contains the live region. This can typically be ignored, this exxists to aid in chaining creation of non-existant regions.
```
```suggestion
 * @return {!Promise<HTMLElement>} a div that contains the live region. This can typically be ignored, this exists to aid in chaining creation of non-existant regions.
```

jcsteh commented on AsusRouterA11yFixes.user.js:
```diff
@@ -0,0 +1,266 @@
+}
+
+/**
+ * create or fetch a live region that can be used with updateLiveRegion. Returns a promise with the region.
```
```suggestion
 * create or fetch a live region that can be used with announce(). Returns a promise with the region.
```

jcsteh commented on framework/axSGreaseSkeleton.js:
```diff
@@ -11,6 +11,57 @@
+}
+
+/**
+ * create or fetch a live region that can be used with updateLiveRegion. Returns a promise with the region.
```
```suggestion
 * create or fetch a live region that can be used with announce(). Returns a promise with the region.
```

jcsteh committed:
Typos.
